clang/test/CodeGen/X86/prefetchi-error.c

```diff
@@ -0,0 +1,7 @@
+// RUN: %clang_cc1 %s -ffreestanding -triple=x86_64-unknown-unknown -target-feature +prefetchi  -fsyntax-only -verify
+
+#include <immintrin.h>
+
+void test_invalid_prefetchi(void* p) {
+  __builtin_ia32_prefetchi(p, 1); // expected-error {{argument value 1 is outside the valid range [2, 3]}}
```

Comment:
We have a lower and upper range, testing should validate both. Tests should be as complete, especially when it is apparent and to prevent regression in behavior in the future due to new changes.